src/include/enclave/enclave_state.h

```diff
@@ -0,0 +1,59 @@
+#ifndef ENCLAVE_STATE_H
+#define ENCLAVE_STATE_H
+
+#include <shared/sgxlkl_enclave_config.h>
+#include <shared/shared_memory.h>
+
+/* Indicates different states during sgxlkl startup sequence */
+typedef enum sgxlkl_libc_state
+{
+    libc_not_started = 0,
+    libc_initializing = 1,
+    libc_initialized = 2
+} sgxlkl_libc_state_t;
+
+typedef struct sgxlkl_enclave_disk_state
+{
+    size_t host_disk_index;
+    int fd;
+    size_t capacity;
+    bool mounted;
+} sgxlkl_enclave_disk_state_t;
+
+typedef struct
+{
+    uint32_t argc;
+    char** argv;
+    char** envp;
+    Elf64_auxv_t** auxv;
+} elf64_stack_t;
```

Comment:
I'm a big fan of comments.

Reply:
This looks as if it's supposed to be an ELF ABI stack layout, but it isn't actually one.

Reply:
Great, than I'll leave that to you! :-P

Reply:
@davidchisnall That's correct, it only looks like one. I'll leave the details to someone who knows what it's actually supposed to behave like. (See also the corresponding function that compiles this.)

Reply:
(Comments added, marking as resolved.)